test/Microsoft.ML.Tests/TrainerEstimators/CalibratorEstimators.cs

```diff
@@ -0,0 +1,153 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Calibrator;
+using Microsoft.ML.Core.Data;
+using Microsoft.ML.Data;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.Internal.Calibration;
+using Microsoft.ML.Runtime.Internal.Internallearn;
+using Microsoft.ML.Runtime.Training;
+using Microsoft.ML.Trainers;
+using Xunit;
+
+namespace Microsoft.ML.Tests.TrainerEstimators
+{
+    public partial class TrainerEstimators
+    {
+        /// <summary>
+        /// OVA and calibrators
+        /// </summary>
+        [Fact]
+        public void PlattCalibratorEstimator()
+        {
+            var calibratorTestData = GetCalibratorTestData();
+
+            // platCalibrator
+            var platCalibratorEstimator = new PlattCalibratorEstimator(Env, calibratorTestData.transformer.Model, "Label", "Features");
+            var platCalibratorTransformer = ((ITrainerEstimator<CalibratorTransformer<PlattCalibrator>, PlattCalibrator>)platCalibratorEstimator).Fit(calibratorTestData.scoredData);
+
+            //testData
+            checkValidCalibratedData(calibratorTestData.scoredData, platCalibratorTransformer);
+
+            //test estimator
+            //TestEstimatorCore(platCalibratorEstimator, calibratorTestData.scoredData);
+
+            Done();
+        }
+
+        /// <summary>
+        /// OVA and calibrators
+        /// </summary>
+        [Fact]
+        public void FixedPlatCalibratorEstimator()
+        {
+            var calibratorTestData = GetCalibratorTestData();
+
+            // fixedPlatCalibrator
+            var fixedPlatCalibratorEstimator = new FixedPlattCalibratorEstimator(Env, calibratorTestData.transformer.Model, new FixedPlattCalibratorTrainer.Arguments(), "Label", "Features");
+            var fixedPlatCalibratorTransformer = ((ITrainerEstimator<CalibratorTransformer<PlattCalibrator>, PlattCalibrator>)fixedPlatCalibratorEstimator).Fit(calibratorTestData.scoredData);
+
+            checkValidCalibratedData(calibratorTestData.scoredData, fixedPlatCalibratorTransformer);
+
+            //test estimator
+            //TestEstimatorCore(calibratorTestData.pipeline, calibratorTestData.data);
+
+            Done();
+        }
+
+        /// <summary>
+        /// OVA and calibrators
+        /// </summary>
+        [Fact]
+        public void NaiveCalibratorEstimator()
```

Comment:
>NaiveCalibratorEstimator [](start = 20, length = 24)

It generates only 1 and 0 values.

Reply:
This seems fine now. 

---
In reply to: [239148679](https://github.com/dotnet/machinelearning/pull/1816#discussion_r239148679) [](ancestors = 239148679)